lib/mongoid/attributes.rb

```diff
@@ -177,8 +177,14 @@ def write_attribute(name, value)
             attribute_will_change!(field_name)
           end
           if localized
-            attributes[field_name] ||= {}
-            attributes[field_name].merge!(typed_value)
+            present = fields[field_name].try(:localize_present?)
+            loc_key, loc_val = typed_value.to_a.flatten
```

Comment:
If this does what I think it is attempting to do, I suggest `typed_value.first` instead of `typed_value.to_a.flatten`. If I am mistaken what is the purpose of `flatten` here?

Reply:
TIL first is a method on hashes. Much nicer. Thanks.

Reply:
done